src/elements/ExactCFbend.H

```diff
@@ -0,0 +1,665 @@
+                            + 1_prt/192_prt*(3_prt + 8_prt*powi<4>(y_scl) + 12_prt*rho*rho
+                            - 15_prt*powi<4>(rho) + 24_prt*y_scl*y_scl*(-1_prt + rho*rho)
+                            +12_prt*rho*rho*(2_prt - 4_prt*y_scl*y_scl + rho*rho)*ln_rho)*ks;
+                     break;
+                // default:
+                //   TODO: print a warning here that order > 5 is not currently supported
+               }
+
+            }
+
+            // advance momentum by a step of length tau
+            pxout = pxout + tau * dpx;
+            pyout = pyout + tau * dpy;
+
+            //update the particle coordinates
+            particle(1) = xout;
+            particle(2) = pxout;
+            particle(3) = yout;
+            particle(4) = pyout;
+            particle(5) = tout;
+            particle(6) = ptout;
+
+            zeval += tau;
+        }
+
+
+        /** This pushes the reference particle.
+         *
+         * @param[in,out] refpart reference particle
+         */
+        AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+        void operator() (RefPart & AMREX_RESTRICT refpart) const
+        {
+            using namespace amrex::literals; // for _rt and _prt
+
+            // assign input reference particle values
+            amrex::ParticleReal const x = refpart.x;
+            amrex::ParticleReal const px = refpart.px;
+            amrex::ParticleReal const y = refpart.y;
+            amrex::ParticleReal const py = refpart.py;
+            amrex::ParticleReal const z = refpart.z;
+            amrex::ParticleReal const pz = refpart.pz;
+            amrex::ParticleReal const t = refpart.t;
+            amrex::ParticleReal const pt = refpart.pt;
+            amrex::ParticleReal const s = refpart.s;
+
+            // access reference particle values to find relativistic factors
+            //m_beta = refpart.beta();
+            //m_ibeta = 1_prt / m_beta;
+            //m_ibetgam2 = 1_prt / powi<2>(refpart.beta_gamma());
+            amrex::ParticleReal const brho = refpart.rigidity_Tm();
+
+#if AMREX_DEVICE_COMPILE
+            amrex::ParticleReal* k_normal = m_k_normal_d_data;
+#else
+            amrex::ParticleReal* k_normal = m_k_normal_h_data;
+#endif
+            // find magnetic field and curvature; normalize units to MAD-X convention if needed
+            amrex::ParticleReal const curvature = m_unit == 1 ? k_normal[0] / brho : k_normal[0];
+            amrex::ParticleReal const rc = curvature == 0.0 ? 0.0_prt : 1.0_prt/curvature;
+
+            // length of the current slice
+            amrex::ParticleReal const slice_ds = m_ds / nslice();
+
+            // special case of zero field = an exact drift
+            if (rc==0_prt) {
+               // advance position and momentum (drift)
+               amrex::ParticleReal const step = slice_ds /std::sqrt(amrex::Math::powi<2>(pt)-1.0_prt);
```

Comment:
Just some readability
```suggestion
               amrex::ParticleReal const step = slice_ds / std::sqrt(amrex::Math::powi<2>(pt) - 1.0_prt);
```